src/libstd/panic.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Panic support in the standard library
+
+#![unstable(feature = "std_panic", reason = "module recently added",
+            issue = "27719")]
+
+use cell::UnsafeCell;
+use ops::{Deref, DerefMut};
+use ptr::Unique;
+use sync::{Mutex, RwLock};
+use sys_common::unwind;
+use thread::Result;
+
+/// A marker trait which represents "panic safe" types in Rust.
+///
+/// This trait is implemented by default for many types and behaves similarly in
+/// terms of inference of implementation to the `Send` and `Sync` traits. The
+/// purpose of this trait is to encode what types are safe to cross a `recover`
+/// boundary with no fear of exception safety. For more information about
```

Comment:
The switch of terminology to "exception safety" here is a bit confusing.

I'd also prefer to try to inline a basic explanation of this trait in the documentation, in addition to pointing to the RFC. Perhaps a brief mention that Rust in general makes it difficult to observe state shared across "unwinding isolation boundaries" (like threads or `recover`) because such state is often left in an inconsistent state, but `PanicSafe` tells you that a given piece of state will be treated sufficiently transactionally.